spec/compiler/completeness_spec.cr

```diff
@@ -32,6 +32,58 @@ describe Mare::Compiler::Completeness do
     end
   end
 
+  it "complains when field write is in while loop in constructor" do
+    source = Mare::Source.new_example <<-SOURCE
+    :class Data
+      :prop w U64
+      :new
+        while True (
+          @w = 0
+        )
+    SOURCE
+
+    expected = <<-MSG
+    This field may be read before it is initialized by a constructor:
+    from (example):2:
+      :prop w U64
+            ^
+
+    - traced from a call here:
+      from (example):5:
+        @w = 0
+        ^~~~~~
```

Comment:
This error message is intended to represent that the field may be read during the constructor before it is first initialized, so it doesn't make very much sense to use it for this case (where the field is never read - it is only written, but conditionally so).

I would expect the error to look like this:
```suggestion
    This constructor doesn't initialize all of its fields:
    from (example):3:
      :new
       ^~~

    - this field didn't get initialized:
      from (example):2:
      :prop w U64
            ^

    - it is initialized here, but this expression is not guaranteed to execute:
      from (example):5:
        @w = 0
        ^~~~~~
```

Reply:
I don't know why this is resolved as FieldRead
=